onnx/defs/shape_inference.h

```diff
@@ -38,7 +38,7 @@ class SymbolTable {
   // Adds existing symbols from a main graph or subgraph
   virtual void addFromGraph(const GraphProto& g) = 0;
   // Creates a new symbol which is not duplicate as any existing one
-  virtual std::string createNew(const std::string& symbol_prefix) = 0;
+  virtual std::string createNew(const std::string& symbol_prefix = "unk__") = 0;
```

Comment:
Still this warning:

warning: default arguments on virtual or override methods are prohibited [google-default-arguments] 

```cpp virtual std::string createNew(const std::string& symbol_prefix = "unk__") = 0; ^
```

Reply:
https://google.github.io/styleguide/cppguide.html#Default_Arguments

> Default arguments are banned on virtual functions, where they don't work properly, and in cases where the specified default might not evaluate to the same value depending on when it was evaluated. (For example, don't write void f(int n = counter++);.)

> In some other cases, default arguments can improve the readability of their function declarations enough to overcome the downsides above, so they are allowed. When in doubt, use overloads.

@gramalingam @jcwchen should we use overloads then?

Reply:
Maybe we can rename the virtual function to something else, say `createNewImpl`, and define `createNew` as a non-virtual function in the base-class, with default-parameter, that calls `createNewImpl`?

Reply:
Consulted with Ed and agreed that either

1. What Rama suggested "rename to createNewImpl, and define createNew as a non-virtual function in the base-class, with default-parameter, that calls createNewImpl?"
2. Define a non-virtual overload `createNew`

will work.

Reply:
Renamed the existing virtual method to `createNewImpl()`, which is now called by the non-virtual `createNew()` method in the base class, with the default parameter available for `createNew()`.